ocean_lib/ocean/ocean_compute.py

```diff
@@ -223,124 +45,49 @@ def _status_from_job_info(job_info: Dict[str, Any]) -> Dict[str, Any]:
             "statusText": job_info["statusText"],
         }
 
-    # can not type check due to Subscripted generics error
-    @staticmethod
-    def check_output_dict(
-        output_def: Optional[Dict[str, Any]],
-        consumer_address: str,
-        data_provider: DataServiceProvider,
-        config: Config,
-    ) -> Dict[str, Any]:
-        """
-        Validate the `output_def` dict and fills in defaults for missing values.
-
-        :param output_def: dict
-        :param consumer_address: hex str the consumer ethereum address
-        :param data_provider:  DataServiceProvider class or similar interface
-        :param config: Config instance
-        :return: dict a valid `output_def` object
-        """
-        default_output_def = {
-            "nodeUri": config.network_url,
-            "brizoUri": data_provider.get_url(config),
-            "brizoAddress": config.provider_address,
-            "metadata": dict(),
-            "metadataUri": config.metadata_cache_uri,
-            "owner": consumer_address,
-            "publishOutput": 0,
-            "publishAlgorithmLog": 0,
-            "whitelist": [],
-        }
-
-        output_def = output_def if isinstance(output_def, dict) else dict()
-        default_output_def.update(output_def)
-        return default_output_def
-
     @enforce_types
     def start(
         self,
-        input_datasets: list,
+        provider_uri: str,
```

Comment:
why do we need the provider_uri? We have service endpoints on the services, as far as I know. The DataService provider should build the provider_uri based on the service endpoint.

Reply:
You're right. good catch!  Done.